pallets/ddc-clusters/src/lib.rs

```diff
@@ -106,7 +119,21 @@ pub mod pallet {
 				.map_err(|_| Error::<T>::AttemptToAddNonExistentNode)?;
 			ensure!(node.get_cluster_id().is_none(), Error::<T>::NodeIsAlreadyAssigned);
 
-			// todo: check that node is authorized by the 'NodeProviderAuthSC' contract
+			let is_authorized: bool = pallet_contracts::Pallet::<T>::bare_call(
+				caller_id,
+				cluster.extension_smart_contract,
+				Default::default(),
+				Default::default(),
+				None,
+				Vec::from([0x96, 0xb0, 0x45, 0x3e]), // blake2("is_authorized"), https://use.ink/basics/selectors#selector-calculation
+				false,
```

Comment:
How about using a [constant selector attribute](https://use.ink/macros-attributes/selector) in the SC interface? This way we would not be dependent on the method name in SCs which is good as our the SC interface and Protocol may evolve and change the original naming in the future

Reply:
Sure. As far as I understand, using the `#[ink(selector = 0x96b0453e)]` on the message handler an authorization extension contract developer can already use any function here no matter what the name. Please confirm or describe what should be changed.

Reply:
If more flexibility is required, we can get the selector from the runtime composition (by adding the type parameter to the `Config`) or from the user together with the contract address.

Reply:
I see, if the `#[ink(selector = 0x96b0453e)]` for the authorization method is what we intended to use in the `NodeProviderSC`, then ok. I supposed that this selector was auto-generated by the ink! from the `is_authorized` method name, and in case we change the name, the selector becomes invalid. But if you are aware of the constant selector then everything is good. I don't think we need to accept the selector from users dynamically, let's just use a constant one. The Pallet's `Config` or just a `const` - both work for me

Reply:
Updated